src/main/java/com/cupid/jikting/member/service/CustomOAuth2UserService.java

```diff
@@ -0,0 +1,74 @@
+package com.cupid.jikting.member.service;
+
+import com.cupid.jikting.member.entity.Member;
+import com.cupid.jikting.member.entity.SocialType;
+import com.cupid.jikting.member.oauth2.CustomOAuth2User;
+import com.cupid.jikting.member.oauth2.OAuthAttributes;
+import com.cupid.jikting.member.repository.MemberRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.Map;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class CustomOAuth2UserService implements OAuth2UserService<OAuth2UserRequest, OAuth2User> {
+
+    private final MemberRepository memberRepository;
+
+    private static final String KAKAO = "kakao";
+
+    @Override
+    public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
+        log.info("CustomOAuth2UserService.loadUser() 실행 - OAuth2 로그인 요청 진입");
+        OAuth2UserService<OAuth2UserRequest, OAuth2User> delegate = new DefaultOAuth2UserService();
+        OAuth2User oAuth2User = delegate.loadUser(userRequest);
+        String registrationId = userRequest.getClientRegistration().getRegistrationId();
+        SocialType socialType = getSocialType(registrationId);
+        String userNameAttributeName = userRequest.getClientRegistration()
+                .getProviderDetails()
+                .getUserInfoEndpoint()
+                .getUserNameAttributeName();
+        Map<String, Object> attributes = oAuth2User.getAttributes();
+        OAuthAttributes extractAttributes = OAuthAttributes.of(socialType, userNameAttributeName, attributes);
+        Member createdMember = getMember(extractAttributes, socialType);
```

Comment:
p2. 현재 변수에 너무 담는 로직이 너무 많아서 오히려 코드가 한 눈에 들어오지 않는 것 같아요. 아래 방법 외에도 어떤 식으로든 코드를 좀 정리할 수 있을 것 같아요!

```suggestion
        OAuth2UserService<OAuth2UserRequest, OAuth2User> delegate = new DefaultOAuth2UserService();
        String userNameAttributeName = userRequest.getClientRegistration()
                .getProviderDetails()
                .getUserInfoEndpoint()
                .getUserNameAttributeName();
        OAuthAttributes extractAttributes = OAuthAttributes.of(
                getSocialType(userRequest.getClientRegistration().getRegistrationId()),
                userNameAttributeName,
                delegate.loadUser(userRequest).getAttributes());
        Member createdMember = getMember(extractAttributes, socialType);
```

Reply:
변수들이 두 번씩 쓰여서 어쩔수가 없는것 같습니다!